src/glados/engine/tui_integration.py

```
@@ -0,0 +1,193 @@
+    def unregister_tui_callback(self, event_type: str) -> None:
+        """Unregister a TUI callback for the specified event type.
+        
+        Args:
+            event_type: Type of event to stop listening for
+        """
+        with self._tui_update_lock:
+            if event_type in self.tui_callbacks:
+                del self.tui_callbacks[event_type]
+                logger.debug(f"Unregistered TUI callback for event type: {event_type}")
+
+    def notify_tui(self, event_type: str, data: Dict[str, Any]) -> None:
```

Ruff (0.11.9)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)